### PR TITLE
[TEST] fix tests

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: theano/sandbox/cuda/opt.py [hunk 1]
```diff
@@ -1785,8 +1785,10 @@ def local_gpu_downsample_factor_max_grad(node):
...
     if isinstance(node.op, downsample.DownsampleFactorMaxGradGrad):
-        assert node.op.__props__ == ('ds', 'ignore_border', 'st')
-
+        assert node.op.__props__ == ('ds', 'ignore_border', 'st',
+                                     'padding', 'mode')
+        if node.op.padding != (0, 0) or node.op.mode != 'max':
+            return
         x, z, gx = node.inputs
         if (x.owner and isinstance(x.owner.op, HostFromGpu)):
             op = GpuDownsampleFactorMaxGradGrad(node.op.ds,
```
Comment (lamblin) [inline theano/sandbox/cuda/opt.py]: We need to pass the new props here as well...

File: theano/sandbox/test_multinomial.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ def get_mode(gpu):
                               'local_gpu_multinomial')
     if isinstance(mode.linker, theano.gof.PerformLinker):
         mode.linker = predefined_linkers['c|py']
+    if not getattr(mode.linker, 'c_thunks', True):
+        mode.linker.c_thunks = True
```
Comment (lamblin) [inline theano/sandbox/test_multinomial.py]: Do all linkers have a `c_thunks` attribute?

File: theano/tensor/tests/test_raw_random.py [hunk 3]
```diff
@@ -1189,12 +1189,15 @@ def test_pkl(self):
...
+        if theano.config.mode in ["DEBUG_MODE", "DebugMode"]:
+            mode = 'FAST_COMPILE'
         post_bin_r, bin_sample = binomial(rng_r, (3, 5), 1, .3)
-        f = theano.function([rng_r], [post_bin_r, bin_sample])
+        f = theano.function([rng_r], [post_bin_r, bin_sample], mode=mode)
         pkl_f = pickle.dumps(f)
 
         post_int_r, int_sample = random_integers(rng_r, (3, 5), -1, 8)
-        g = theano.function([rng_r], [post_int_r, int_sample])
+        g = theano.function([rng_r], [post_int_r, int_sample], mode=mode)
```
Comment (lamblin) [inline theano/tensor/tests/test_raw_random.py]: Right, I forgot about that case. Thanks.

File: theano/sandbox/cuda/opt.py (hunk 1 above)
Comment (lamblin) [inline theano/sandbox/cuda/opt.py]: Oh, I did not see that we abort the optimization for unsupported versions of padding and mode.
However, "st" should not be ignored.

File: theano/sandbox/test_multinomial.py (hunk 2 above)
Comment (nouiz) [inline theano/sandbox/test_multinomial.py]: no, only VM_Linker.

File: theano/sandbox/cuda/opt.py (hunk 1 above)
Comment (nouiz) [inline theano/sandbox/cuda/opt.py]: I updated the PR.

File: theano/sandbox/test_multinomial.py (hunk 2 above)
Comment (lamblin) [inline theano/sandbox/test_multinomial.py]: Then, maybe we should not set it to `True` if it did not exist.

Comment (nouiz) [inline theano/sandbox/test_multinomial.py]: It is a copy, so we don't care, but I changed it (and added a small commit)

**Pull request merged**